From 5763a9a76b98662dc94278bc2cecc9fae8bd18b4 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 7 Mar 2018 11:14:53 +0800
Subject: [PATCH 3440/5242] MLK-17156-7: ASoC: imx-rpmsg: use rpmsg codec
 instead the dummy

commit  69fbd0770654d2642e180b1f6e8785caa88c8a50 from
https://source.codeaurora.org/external/imx/linux-imx.git

use the rpmsg_wm8960 codec instead of the dummy codec

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/Kconfig     |    1 +
 sound/soc/fsl/imx-rpmsg.c |    4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/Kconfig b/sound/soc/fsl/Kconfig
index 9624431..df97a99 100644
--- a/sound/soc/fsl/Kconfig
+++ b/sound/soc/fsl/Kconfig
@@ -402,6 +402,7 @@ config SND_SOC_IMX_RPMSG
 	depends on OF && I2C && INPUT
 	select SND_SOC_IMX_PCM_RPMSG
 	select SND_SOC_FSL_RPMSG_I2S
+	select SND_SOC_RPMSG_WM8960
 	help
 	  SoC Audio support for i.MX boards with rpmsg.
 	  There should be rpmsg devices defined in other core
diff --git a/sound/soc/fsl/imx-rpmsg.c b/sound/soc/fsl/imx-rpmsg.c
index 72b2e01..6887e69 100644
--- a/sound/soc/fsl/imx-rpmsg.c
+++ b/sound/soc/fsl/imx-rpmsg.c
@@ -58,8 +58,8 @@ static int imx_rpmsg_probe(struct platform_device *pdev)
 
 	data->dai[0].name = "rpmsg hifi";
 	data->dai[0].stream_name = "rpmsg hifi";
-	data->dai[0].codec_dai_name = "snd-soc-dummy-dai";
-	data->dai[0].codec_name = "snd-soc-dummy";
+	data->dai[0].codec_dai_name = "rpmsg-wm8960-hifi";
+	data->dai[0].codec_name = "rpmsg-audio-codec";
 	data->dai[0].cpu_dai_name = dev_name(&cpu_pdev->dev);
 	data->dai[0].platform_of_node = cpu_np;
 	data->dai[0].playback_only = false;
-- 
1.7.9.5

